Description: on NetBSD and OpenBSD, $bootdir is empty by default.  Hence,
  grub-install runs mkdir "", which fails.  This patch fixes this problem.
  Moreover, the patch also permits a root directory of arbitrary depth.

--- util/grub-install.in.orig	2010-01-02 14:42:38.000000000 +0100
+++ util/grub-install.in
@@ -207,8 +207,7 @@ else
 fi
 
 # Create the GRUB directory if it is not present.
-test -d "$bootdir" || mkdir "$bootdir" || exit 1
-test -d "$grubdir" || mkdir "$grubdir" || exit 1
+mkdir -p "$grubdir" || exit 1
 
 # If --recheck is specified, remove the device map, if present.
 if test $recheck = yes; then
--- util/i386/efi/grub-install.in.orig	2010-01-02 14:42:38.000000000 +0100
+++ util/i386/efi/grub-install.in
@@ -143,8 +143,7 @@ else
 fi
 
 # Create the GRUB directory if it is not present.
-test -d "$bootdir" || mkdir "$bootdir" || exit 1
-test -d "$grubdir" || mkdir "$grubdir" || exit 1
+mkdir -p "$grubdir" || exit 1
 
 # If --recheck is specified, remove the device map, if present.
 if test $recheck = yes; then
--- util/ieee1275/grub-install.in.orig	2010-01-02 14:42:38.000000000 +0100
+++ util/ieee1275/grub-install.in
@@ -140,8 +140,7 @@ fi
 # XXX warn on firmware-unreadable filesystems?
 
 # Create the GRUB directory if it is not present.
-test -d "$bootdir" || mkdir "$bootdir" || exit 1
-test -d "$grubdir" || mkdir "$grubdir" || exit 1
+mkdir -p "$grubdir" || exit 1
 
 # Create the device map file if it is not present.
 if test -f "$device_map"; then
